src/main/java/org/pih/petl/api/JobExecutor.java

```diff
@@ -50,59 +81,95 @@ public void executeInParallel(List<JobExecutionTask> tasks) throws InterruptedEx
         while (tasksToSchedule.size() > 0) {
             List<Future<JobExecutionResult>> futures = new ArrayList<>();
             for (JobExecutionTask task : tasksToSchedule) {
+                JobExecution execution = task.getExecutionContext().getJobExecution();
```

Comment:
Most of the remaining changes to the JobExecutor are in ensuring that job executions have their status updated at appropriate times in the database to facilitate tracking status by end users.